server/data/restClient.ts

```diff
@@ -8,23 +8,23 @@ import { ApiConfig } from '../config'
 import type { UnsanitisedError } from '../sanitisedError'
 
 interface GetRequest {
-  path?: string
+  path: string
```

Comment:
That was a weird one, it seems like the idea is that if you don't pass `path` it would make a request to just `this.apiUrl()` (because of this ``(`${this.apiUrl()}${path}`)``)

So in theory making this mandatory could break the client code if there are any calls which don't pass `path`? I think.

I still find default to `null` a bit weird given this gets interpolated into a string, maybe it should still be optional but default to an empty string?

I'm probably missing something here.

Reply:
oh good spot. perhaps i should allow it to be `undefined` and coerce it like `path = path || '';`

in the past, if one didn't provide a path, the bit you highlight would actually say "null" in place of path so it was still not quite right. lemme take another look…

Reply:
This is exactly why defaulting to `null` makes me a bit nervous - it would be interpolated in a string and we all know how lovely these strings with `undefined` in it are 🤣  (but maybe `null` is "special"/different?)

I would still prefer to be more explicit, we could default to an empty string right? You make the type of path `string` and it defaults to `''`, win-win.

NOTE: It doesn't look like `null` is special when interpolating strings 😬 : https://jsfiddle.net/muxj7nkd/ - I imagine this means all client code is passing `path` explicitly?